unit-tests/func/reflectivity/l500/test-auto-set-options.cpp

```diff
@@ -0,0 +1,192 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+//#test:tag ir-reflectivity
+//#test:device L500*
+
+#include "../../func-common.h"
+
+// Test group description:
+//       * This tests group verifies IR reflectivity option restrictions,
+//         according to [RS-8358].
+//
+// Current test description:
+//       * While not streaming, activating IR Reflectivity option should make sure the visual preset is 'Max Range' and the sensor mode is 'VGA', 
+//         if it is not it will set the visual preset to max range and the sensor mode to 'VGA'.
+//       * Starting non VGA stream when IR Reflectivity is on, turns it off.
+//       * Activating IR Reflectivity while Max Usable Range is off
+//         should turn on Max Usable Range option.
+//         If the MUR option was turned on by activating IR Reflectivity,
+//         Turning off IR Reflectivity should turn off MUR option.
+//       * If MUR option was already on when IR Reflectivity was turned on,
+//         turning off IR Reflectivity should not turn off MUR option.
+//       * While streaming, Activating IR Reflectivity while Max Usable Range is off
+//         should turn on Max Usable Range option.
+//         If the MUR option was turned on by activating IR Reflectivity,
+//         Turning off IR Reflectivity should turn off MUR option.
+//       * Verify turning off MUR option turns off IR Reflectivity option,
+//         Reflectivity require MUR to be on otherwise it can't work.
+//
+// Note: * L515 specific test
+//       * Test will pass if no L515 device is connected.
+//       * Test will be performed on the first L515 device detected.
+
+TEST_CASE( "IR Reflectivity option change visual preset and sensor mode if not streaming", "[reflectivity]" )
+{
+    auto ds
+        = find_first_supported_depth_sensor_or_exit( "L515", RS2_OPTION_ENABLE_IR_REFLECTIVITY );
+
+    // Verify alt IR is off
+    if( ds.supports( RS2_OPTION_ALTERNATE_IR ) )
+        ds.set_option( RS2_OPTION_ALTERNATE_IR, 0 );
+
+    ds.set_option( RS2_OPTION_SENSOR_MODE, RS2_SENSOR_MODE_XGA );
+    ds.set_option( RS2_OPTION_VISUAL_PRESET, RS2_L500_VISUAL_PRESET_SHORT_RANGE );
+    CHECK_NOTHROW( ds.set_option( RS2_OPTION_ENABLE_IR_REFLECTIVITY, 1.f ) );
+    CHECK( RS2_L500_VISUAL_PRESET_MAX_RANGE == ds.get_option( RS2_OPTION_VISUAL_PRESET ) );
+    CHECK(RS2_SENSOR_MODE_VGA == ds.get_option(RS2_OPTION_SENSOR_MODE));
```

Comment:
For MUR this test fell under `test-dependencies.cpp` -- up to you

Reply:
renamed to test-dependencies for alignment with MUT tests